### DISCUSSION TITLE
GraphQL search results ordering broken

### DISCUSSION THREAD

**Discussion by nettrino** [community/community#51660] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

Hello all, 

I was trying to fetch pull requests via GraphQL and noticed that the search results appear in the wrong order.
In the following query, it does not matter if I do `first` or `last`, the same results are always returned, and if I try pagination then I don't get the same output as when I look at the UI in github. 
```graphql
{
  search(
    query: "is:merged is:pr repo:<sample_repo> user:<sample_org>"
    type: ISSUE
    first: 5
  ) {
    pageInfo {
      hasPreviousPage
      hasNextPage
      endCursor
      startCursor
    }
    nodes {
      ... on PullRequest {
        id
        number
      }
    }
  }
}
```

Am I doing something wrong or is this a bug?

**Comment by Gisgar3** [comment, marked as answer]
Hey there, welcome to the **GitHub Community Forum**! I ran this query through GitHub's GraphQL Explorer and am having the same issue too; not sure if this happens to be a bug or not within this context. However, I tinkered around with it a bit more and got it to properly utilize the ``first`` and ``last`` parameters when calling it this way:
```graphql
query {
  repository(name: "<sample-repo>", owner: "<sample-owner>") {
    pullRequests(first: 5, states: MERGED) {
      pageInfo {
        hasPreviousPage
        hasNextPage
        endCursor
        startCursor
      }
      edges {
        node {
          id
          number
        }
      }
    }
  }
}
```

I hope this helps you out and is what you were looking for! If you have anymore questions, do not hesitate to ask away 👍🏻 :octocat:

**Comment by nettrino** [comment]
Thank you for the reply - I had switched to the workaround as well but curious to know what exactly is the issue with the search functionality as there might be scenarios where that is the only option. The nice thing about the current search results is that they do return somewhat arbitratry results from the PRs, which actually could be useful for users. The API supports forward/backwards pagination but no "random" ordering as far as I know correct?

**Reply by Gisgar3** [reply]
I'm as curious as you are about the search functionality issue too. Not exactly sure why this would be occurring. In regards to pagination, the 'PageInfo' object seems to only have options for confirming if there's more items going forward or backwards in paginating, and the end and start cursors for paginating forwards or backwards respectively. I am glad that the workaround is working good for you however! If you can mark my first reply as a valid answer, that'll help verify that the initial code issue is fixed! I'm going to tag GitHub staff to see if they can relay this thread and get an explanation for the search issue 👍🏻

@queenofcorgis
